Kafka 4.0 (Kraft only) support

ShvaykaD opened the pull request:
![image](https://github.com/user-attachments/assets/b529b166-4c5e-40e8-86eb-1d4c5d7f04c8)
![image](https://github.com/user-attachments/assets/c01d3a46-695c-4245-9d93-50969f24c296)

smatvienko-tb commented on kafka.env:
```diff
@@ -1,11 +1,19 @@
 KAFKA_CFG_AUTO_CREATE_TOPICS_ENABLE=false
-KAFKA_CFG_INTER_BROKER_LISTENER_NAME=INSIDE
 KAFKA_CFG_LOG_RETENTION_BYTES=1073741824
+# Kafka default value: 1 GB. TB value: ~27MB
```
actually I see ~250MB
Probably here is an old mistake (not yours) same as on CE

ShvaykaD replied:
@smatvienko-tb what do you think is better in this case? Reset the value to ~27MB or update the comment back to ~250MB?

smatvienko-tb replied:
@ShvaykaD please take a look on the number it Is one digit more than appears.
268435456/1024/1024=256 MB

smatvienko-tb replied:
I have already changed to 25 MB on CE

smatvienko-tb reviewed (approved):
Originally from CE https://github.com/thingsboard/thingsboard/pull/13223/files

I believe you gave a chance to run the docker compose on PE.

On the CE the tests performed on build machine. But for PE it is not happened until the submodule changed in thingsboard/PE. And in PE a lot more options we have to deploy (3 instead of 1 for CE).

If so, please, provide a Kafka container output in the PR description. 

Without an output I can only speculate and assuming but not 100% sure that is it working fine

smatvienko-tb commented:
CE link : https://github.com/thingsboard/thingsboard/pull/13223

ShvaykaD commented:
Hi @smatvienko-tb, sure, I have run the PE version after adding my changes. See my inline comments below:

> Originally from CE [thingsboard/thingsboard#13223](https://github.com/thingsboard/thingsboard/pull/13223)
> 
> I believe you had a chance to run the docker compose on PE.
> 

The only comment that I have run only a "basic" TB setup. I don't think it makes a lot of sense to run all the setups, as we changed only the Kafka itself. The way TB communicates with Kafka remained untouched.

> On the CE the tests performed on build machine. But for PE it is not happened until the submodule changed in thingsboard/PE. And in PE a lot more options we have to deploy (3 instead of 1 for CE).
> 
> If so, please, provide a Kafka container output in the PR description.
> 
> Without an output I can only speculate and assuming but not 100% sure that is it working fine

Added screenshots to the PR description.

smatvienko-tb reviewed (approved):
gtm
